feat: add Info properties to Intl.Locale (Calendars) (#8768)
mdn/content: files/en-us/web/javascript/reference/global_objects/intl/locale/calendars/index.md

---
title: Intl.Locale.prototype.calendars
slug: Web/JavaScript/Reference/Global_Objects/Intl/Locale/calendars
tags:
  - Internationalization
  - Intl
  - JavaScript
  - Property
  - Locale
  - Localization
  - Prototype
  - Reference
  - Calendars
browser-compat: javascript.builtins.Intl.Locale.calendars
---
{{JSRef}}

The **`Intl.Locale.prototype.calendars`** property is an accessor property which returns a list of 1 or more unique calendar identifiers for the `Locale`.

## Description

> The `calendar` property returns a list of all supported calendars for the  `Locale`  all list items indicates the `Locale`'s calendar era. The following table shows all the valid Unicode calendar key strings, along with a description of the calendar era they represent.

Reviewer: Two extra spaces, and I don't like _'s_ after code-like phrase. I propose:
```suggestion
The `calendar` property returns a list of all supported calendars for the `Locale` all list items indicates the `Locale` object's calendar era. The following table shows all the valid Unicode calendar key strings, along with a description of the calendar era they represent.
```

### Unicode calendar keys

> <table class="standard-table">

Reviewer: A table is not a good structure for this, a definition list is much better. We use an extension of Github Markdown for these ([documentation](https://developer.mozilla.org/en-US/docs/MDN/Contribute/Markdown_in_MDN#definition_lists))

So basically:
```
* `buddhist`
    * : Thai Buddhist calendar
* `chinese`
    * : Traditional Chinese calendar
[…]
```

Author: This table is a copy from https://github.com/mdn/content/blob/cc6d65453242d60e746579de4cecd6892cee57d3/files/en-us/web/javascript/reference/global_objects/intl/locale/calendar/index.md, should both be updated?

Reviewer: Yes, please.

Author: Not questioning but only giving suggestions, IMHO feels more readable and saw it in several places like [Intl.Locale.prototype.numberingSystem
](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/numberingSystem)

![image](https://user-images.githubusercontent.com/1572026/133066044-7a38b317-ba07-4e72-b4b5-8454a6326c2a.png)

Aside from that how I can add those warnings using the list 

![image](https://user-images.githubusercontent.com/1572026/133066147-bd7b17dd-540d-46e4-9922-edbcdc62657c.png)

Author: The same happens here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/collation
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/hourCycle
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/numberingSystem

Those pages I need to replicate to add documentation for finish the Locale Info

Reviewer 2: Hi!

It seems like there are maybe three questions in here, and I'm not sure which is the one I am being asked :).

**1) how can I represent those warnings using the list syntax?**

You can include block elements including warning boxes in list items:

```
- a term
  - : a description of the term
    > **Warning:** It can contain warnings

- another term
   - : another description
 ```

...although in this particular case I might prefer to have a separate section "deprecated collation types" or something, rather than using a warning note.

**2) should we use tables rather than `<dl>`?**

It's **very** conventional on MDN to use `<dl>` for things like the value that a property can take. Consistency is really important for reference docs, so if we want to talk about using tables instead that should be a whole MDN-wide conversation (and honestly I don't think it would get far). We should not even consider doing something different in some small set of cases just because it feels nicer there. Or perhaps I misunderstand this question?

**3) how can we include the same content in multiple places in MDN?**

This is tricky! We have a macro for this  called [`{{page}}`](https://github.com/mdn/yari/blob/main/kumascript/macros/page.ejs). I think this should still work in Markdown (except that you can in general only refer to ID attributes that are generated from headings).

But `{{page}}` is problematic for a couple of reasons. First the implementation is very tricky because it introduces a build dependency between different pages in the site. Second, when used ad hoc it is also  hard to maintain, because the "source" text doesn't know where it is being embedded (or even that it is being embedded at all) so it's very easy for maintainers of the source text to make changes that destroy the meaning of the text in its embedded context.

So we would really like to deprecate the general use of `{{page}}`. Unfortunately though, there is a real use case for it as you point out, and duplicating content across many pages is also bad for maintainability. If possible I like to use links for this, but that's not always possible. I wonder if eventually we might get a more controlled version of `{{page}}`, where the source texts are explicitly identified as fragments (rather than just being normal parts of normal pages). But that seems a long way off.

Author: First thanks for your feedback, regarding this :

> **2) should we use tables rather than `<dl>`?**
> 
> It's **very** conventional on MDN to use `<dl>` for things like the list of values that a property can take. Consistency is really important for reference docs, so if we want to talk about using tables instead that should be a whole MDN-wide conversation (and honestly I don't think it would get far). We should not even consider doing something different in some small set of cases just because it feels nicer there. Or perhaps I misunderstand this question?
> 

My point is that the table syntax is used all over those documents that I'm only duplicating to a new page, if the intention is to have consistency by using `<dl>` where previously we used tables I understand but is a different effort that I wanted to clarify.  Thanks

Author: Updated the PR, thanks for clarifying

---
title: Intl.Locale.prototype.calendars
slug: Web/JavaScript/Reference/Global_Objects/Intl/Locale/calendars
tags:
  - Internationalization
  - Intl
  - JavaScript
  - Property
  - Locale
  - Localization
  - Prototype
  - Reference
  - Calendars
browser-compat: javascript.builtins.Intl.Locale.calendars
---
{{JSRef}}

> The **`Intl.Locale.prototype.calendars`** property is an accessor property which returns a list of 1 or more unique calendar identifiers for the `Locale`.

Reviewer 3: ```suggestion
The **`Intl.Locale.prototype.calendars`** property is an accessor property which returns an array of one or more unique calendar identifiers for the `Locale`.
```

## Description

> The `calendar` property returns a list of all supported calendars for the `Locale` all list items indicates the `Locale` object's calendar era. The following table shows all the valid Unicode calendar key strings, along with a description of the calendar era they represent.

Reviewer 3: ```suggestion
The `calendar` property returns an array of all supported calendars for the `Locale`. The array items indicate the `Locale` object's calendar era. The following table shows all valid Unicode calendar key strings, along with a description of the calendar era they represent.
```

### Unicode calendar keys

- `buddhist`
  - : Thai Buddhist calendar
- `chinese`
  - : Traditional Chinese calendar
- `coptic`
  - : Coptic calendar
- `dangi`
  - : Traditional Korean calendar
- `ethioaa`
  - : Ethiopic calendar, Amete Alem (epoch approx. 5493 B.C.E)
- `ethiopic`
  - : Ethiopic calendar, Amete Mihret (epoch approx, 8 C.E.)
- `gregory`
  - : Gregorian calendar
- `hebrew`
  - : Traditional Hebrew calendar
- `indian`
  - : Indian calendar
- `islamic`
  - : Islamic calendar
- `islamic-umalqura`
  - : Islamic calendar, Umm al-Qura
- `islamic-tbla`
  - : Islamic calendar, tabular (intercalary years [2,5,7,10,13,16,18,21,24,26,29] - astronomical epoch)
- `islamic-civil`	
  - : Islamic calendar, tabular (intercalary years [2,5,7,10,13,16,18,21,24,26,29] - civil epoch)
- `islamic-rgsa`
  - : Islamic calendar, Saudi Arabia sighting
- `iso8601`
  - : ISO calendar (Gregorian calendar using the ISO 8601 calendar week rules)
- `japanese`	
  - : Japanese Imperial calendar
- `persian`
  - : Persian calendar
- `roc`
  - : Civil (algorithmic) Arabic calendar
- `islamicc`
  - : Civil (algorithmic) Arabic calendar

>     > **Warning:** The islamicc calendar key has been deprecated. Please use islamic-civil.

Reviewer 3: ```suggestion
    > **Warning:** The `islamicc` calendar key has been deprecated. Please use `islamic-civil`.
```

## Examples

### Obtaining supported calendars

List supported calendars for a given `Locale`.

```js
let arEG = new Intl.Locale("ar-EG");

> console.log(arEG.calendars); // Prints ["gregory", "coptic", "islamic", "islamic-civil", "islamic-tbla"

Reviewer 3: ```suggestion
console.log(arEG.calendars); // logs ["gregory", "coptic", "islamic", "islamic-civil", "islamic-tbla"]
```

```

```js
let jaJP = new Intl.Locale("ja-JP");

> console.log(jaJP.calendars); // Prints ["gregory", "japanese"]

Reviewer 3: ```suggestion
console.log(jaJP.calendars); // logs ["gregory", "japanese"]
```
